skrub/_agg_joiner.py

```diff
@@ -145,44 +128,166 @@ class AggJoiner(BaseEstimator, TransformerMixin):
     ...     "total_passengers": [90, 120, 100, 70, 80, 90],
     ...     "company": ["DL", "AF", "AF", "DL", "DL", "TR"],
     ... })
-    >>> join_agg = AggJoiner(
+    >>> agg_joiner = AggJoiner(
     ...     aux_table=aux,
-    ...     aux_key="from_airport",
     ...     main_key="airportId",
+    ...     aux_key="from_airport",
     ...     cols=["total_passengers", "company"],
     ...     operation=["mean", "mode"],
     ... )
-    >>> join_agg.fit_transform(main)
-       airportId airportName company_mode_1  total_passengers_mean_1
-    0          1   Paris CDG             AF               103.33...
-    1          2      NY JFK             DL                80.00...
+    >>> agg_joiner.fit_transform(main)
+       airportId  airportName  company_mode  total_passengers_mean
+    0          1    Paris CDG            AF              103.33...
+    1          2       NY JFK            DL               80.00...
     """
 
     def __init__(
         self,
         aux_table,
         *,
-        aux_key,
-        main_key,
+        key=None,
+        main_key=None,
+        aux_key=None,
         cols=None,
         operation=None,
-        suffix=None,
+        suffix="",
     ):
         self.aux_table = aux_table
+        self.key = key
+        self.main_key = main_key
         self.aux_key = aux_key
         self.cols = cols
-        self.main_key = main_key
         self.operation = operation
         self.suffix = suffix
 
+    def _check_dataframes(self, X, aux_table):
+        """Check dataframes input types.
+
+        Raises an error if frames aren't both Pandas or Polars dataframes,
+        or if there is a Polars lazyframe. Alternatively, allows `aux_table` to be "X".
+
+        Parameters
+        ----------
+        X : DataframeLike
+            The main table to augment.
+        aux_table : DataframeLike or "X"
+            The auxiliary table.
+        """
+        if type(aux_table) == str:
+            if aux_table == "X":
+                return X, deepcopy(X)
```

Comment:
why do we need the copy?

Reply:
It was used in the original `AggJoiner`, but we don't need it -> removed

Reply:
Also removed the one in `MultiAggJoiner` at the same time